### PR TITLE
fixed unexpected behavior for opposite parallel planes.

### PR DESCRIPTION
**Pull request by mattiskoh**
`Plane.is_parallel()` does not check for opposite normals and thus returns False for this edge case. 

Not the most elegant fix due to the `TOL.is_close()` hope it's fine like this..

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/geometry/plane.py [hunk 1]
```diff
@@ -384,9 +384,15 @@ def is_parallel(self, other, tol=None):
...
         >>> plane1.is_parallel(plane2)
         True
+        >>> plane1 = Plane.worldXY()
+        >>> plane2 = Plane([1.0, 1.0, 1.0], [0.0, 0.0, -1.0])
+        >>> plane1.is_parallel(plane2)
+        True
 
         """
-        return TOL.is_close(self.normal.dot(other.normal), 1, rtol=0, atol=tol)
+        is_parallel = TOL.is_close(self.normal.dot(other.normal), 1, rtol=0, atol=tol)
```
Comment (tomvanmele) [inline src/compas/geometry/plane.py]: why not just take the absolute value of the dot product?

**Review by tomvanmele — COMMENTED** [review]
thanks for noticing this and proposing a fix...

File: src/compas/geometry/plane.py (hunk 1 above)
Comment (mattiskoh) [reply src/compas/geometry/plane.py]: agreed. I've made the changes.

**Review by tomvanmele — COMMENTED** [review]
the fix is great, but could you add a test that proves the problem is solved?

**Comment by mattiskoh** [comment]
> the fix is great, but could you add a test that proves the problem is solved?

Isn't the Docstring Example sufficient..? Or do you guys not run `doctest` ? Better said why put an example if there is no test for this function...?

**Comment by tomvanmele** [comment]
the examples in the docs are tested. however, examples are not necessarily meant to be formulated as tests, but rather explanations for users on how to use the advertised functionality. in many case some of the examples will indeed overlap with some of the tests, but ideally tests for example also address things like boundary cases, which simple examples usually don't.

currently our test coverage is not nearly sufficient, but we are trying to work towards this. therefore i try to make sure that at least new PRs are accompanied by tests, especially when they fix an issue.

i know it is annoying but it needs to be done...

**Review by tomvanmele — APPROVED** [review]

**Comment by jf---** [comment]
nice find @mattiskoh 

I wonder whether the fix can be generalised; that is, is a tolerance is an unsigned value in general? Intuitively the suggested fix might just be broadly applicable. 

The precision class of OCCT utilises abs in a nr of cases

https://dev.opencascade.org/doc/refman/html/class_precision.html
